$\textbf{Empirical Exercise-Week 4}$  Page 245, Problem 5.28
Let's look at an example about the auction market for art. 

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.api as sm

In [2]:
Tab1 = pd.read_csv('ashcan_small.csv');  sold = Tab1.iloc[:,0]  
Tab1.head(1)

sold  date_auctn  years_old  rhammer  inchsq  signed  creation  age   yob  \
0     1     2008.75     104.75  0.17327   81.04       1      1904   39  1865   

   drec  
0     1

In [3]:
rhammer = Tab1.iloc[:,3];  inchsq = Tab1.iloc[:,4]; yrsold = Tab1.iloc[:,2];  

In [4]:
rhammer1 = np.array(rhammer); inchsq1 = np.array(inchsq); yrsold1 = np.array(yrsold); 
sold1 = np.array(sold);
dummy = (sold1== 1); # only use data for iterms which are sold
rhammer2 = rhammer1[dummy]; inchsq2 = inchsq1[dummy]/10; yrsold2 = yrsold1[dummy]

In [5]:
y = np.log(rhammer2);  xxx = np.hstack([np.ones((len(y),1)), yrsold2.reshape(-1,1), inchsq2.reshape(-1,1)])

In [7]:
result1 = sm.OLS(y,xxx).fit()
print('Coefficient values and t values:\n',np.array([result1.params, result1.tvalues]).T)
print('\n')
print('Coefficient confidence intervals:\n',result1.conf_int())

Coefficient values and t values:
 [[-0.2450809  -0.52293156]
 [ 0.02161639  4.00180299]
 [ 0.02672507  9.71355484]]


Coefficient confidence intervals:
 [[-1.16631288  0.67615108]
 [ 0.01099866  0.03223413]
 [ 0.02131696  0.03213317]]


Answer to question: how much paintings appreciate in one year, between 1.1% and 3.22% at the 95 percent level of confidence

With a 95% confidence interval we can also say that an extra 10 square inches will appreciate a painting betweeh 2.13% and 3.21%. We can reject the null hypothesis that an extra 10 inches will appreciate a painting by 2% in favor of the one-sided alternative that the appreciation will be greater than 2%.

In [6]:
xxxx = np.hstack([np.ones((len(y),1)), yrsold2.reshape(-1,1), inchsq2.reshape(-1,1), (inchsq2**2).reshape(-1,1)])   # Created quadratic term

In [7]:
result2 = sm.OLS(y,xxxx).fit()
result2.conf_int()

array([[-1.86275485e+00, -5.22132742e-02],
       [ 1.28253995e-02,  3.31074092e-02],
       [ 4.58945705e-02,  6.65335797e-02],
       [-2.41952957e-04, -1.29415493e-04]])

The addition of the quadratic term does not change the confidence interval for age.

In [8]:
np.median(inchsq2)

25.40625

Confidence interval for changing by 50, 250, 900 square inches

In [9]:
np.array([result2.conf_int()[1,:]*5 , result2.conf_int()[1,:]*25, result2.conf_int()[1,:]*90])

array([[0.064127  , 0.16553705],
       [0.32063499, 0.82768523],
       [1.15428596, 2.97966683]])

Confidence interval for price-maximizing size of painting: $inchsq2^{*} = .5 * beta2/beta3$

In [10]:
[-0.5*result2.conf_int()[2,0]/result2.conf_int()[3,0] , -0.5*result2.conf_int()[2,1]/result2.conf_int()[3,1]]

[94.84192942998624, 257.05415159617553]

Calculate expected value of auction price for 75 year old and 100 square inch painting and confidence interval

In [11]:
[np.dot([1, 75, 10, 100] , result2.params), np.dot([1,75,10,100] , result2.conf_int()[:,0]), np.dot([1,75,10,100],result2.conf_int()[:,1])]

[1.3085685948409007, -0.4660994766082674, 3.0832366662900688]

Answer is the expected log price and the 95% confidence level. Remember that a negative log value is still a positive price.